### PR TITLE
CI sync

### PR DESCRIPTION
**Pull request by mgates3**
Synchronize CI scripts with SLATE, BLAS++, LAPACK++.
Add tests with Intel compiler. In other projects, this is done for device = gpu_intel, but here there is no device, so just have compiler = intel.

Builds on PR #15, so merge that first.

### CODE REVIEW

File: .github/workflows/setup_env.sh [hunk 1]
```diff
@@ -32,21 +32,38 @@ export top=$(pwd)
...
+export color=no
+export CXXFLAGS="-Werror -Wno-unused-command-line-argument"
+
+#----------------------------------------------------------------- Compiler
+if [ "${compiler}" = "intel" ]; then
+    print "======================================== Load Intel oneAPI compiler"
+    quiet module load intel-oneapi-compilers
+else
+    print "======================================== Load GNU compiler"
+    quiet module load gcc@8.5.0
```
Comment (cayrols) [inline .github/workflows/setup_env.sh:53]: Is it normal to have this module name `gcc@8.5.0`, especially with the symbol @?
Also, does it have to be consistent with all the machines?

File: .github/workflows/test.sh [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
 err=0
 
-cd test
 export OMP_NUM_THREADS=8
-./run_tests.py --xml ${top}/report-${maker}.xml
+
+print "======================================== Tests"
+cd test
+
+./run_tests.py
```
Comment (cayrols) [inline .github/workflows/test.sh:18]: Why do we get rid of the xml file?

**Review by cayrols — APPROVED** [review]
Looks good to me.

File: .github/workflows/setup_env.sh (hunk 1 above)
Comment (mgates3) [reply .github/workflows/setup_env.sh:53]: Yes, it's normal: https://modules.readthedocs.io/en/stable/modulefile.html#version-specifiers
All machines load the same Docker container, so they have the same versions available.

File: .github/workflows/test.sh (hunk 2 above)
Comment (mgates3) [reply .github/workflows/test.sh:18]: It was used by Jenkins. At least right now, it isn't used by GitHub.

**Pull request merged**